backend/src/main/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhir.java

```diff
@@ -292,6 +294,16 @@ private Bundle convertRowToFhirBundle(TestResultRow row, UUID orgId) {
             uuidGenerator.randomUUID().toString(),
             uuidGenerator.randomUUID().toString());
 
+    LocalDateTime testResultDate;
+    TemporalAccessor temporalAccessor =
+        dateTimeFormatter.parseBest(
+            row.getTestResultDate().getValue(), LocalDateTime::from, LocalDate::from);
+    if (temporalAccessor instanceof LocalDateTime) {
+      testResultDate = (LocalDateTime) temporalAccessor;
+    } else {
+      testResultDate = ((LocalDate) temporalAccessor).atStartOfDay();
+    }
+
```

Comment:
This is an awesome way to handle the date parsing! 💯💯💯 I'll update #5985 with this.

Do you think we should move this parsing towards the top of `convertRowToFhirBundle` to group it near `testEventId` since `testResultDate` would now be used by at least `observation` and `diagnosticReport` (I think we might soon be using it for `specimen.collected` too)